modules/waypoint_tracking.py

```diff
@@ -34,19 +35,19 @@ def get_current_waypoint_info(
     # Get the current destination
     if current_waypoint < drone.commands.count:
         current_command = drone.commands[current_waypoint]
-        if current_command.command == dronekit.mavutil.mavlink.MAV_CMD_NAV_WAYPOINT:
+        if current_command.command == drone.is_drone_destination_final_waypoint():
```

Comment:
look at how drone.is_drone_destination_final_waypoint() is implemented in 

https://github.com/UWARG/common/blob/main/mavlink/modules/flight_controller.py

from there you should realize why this code shouldnt be written like this. i can clarify on path optimization channel if needed but you should try and figure it out by looking through the flightcontroller